pkgs/applications/networking/insync/default.nix

```
@@ -38,8 +30,85 @@ stdenv.mkDerivation rec {
      and built in sharing.
 
      There is a 15-day free trial, and it is a paid application after that.
+
+     Known bug(s):
+
+     1) Currently the system try icon does not render correctly.
+     2) libqtvirtualkeyboardplugin does not have necessary Qt library shipped from vendor.
     '';
-    # download URL removed
-    broken = true;
   };
+  insync-pkg = stdenv.mkDerivation {
+    inherit pname;
+    inherit version;
+    inherit meta;
+
+    src = fetchurl {
+      # Find a binary from https://www.insynchq.com/downloads/linux#ubuntu.
+      url = "https://cdn.insynchq.com/builds/linux/${pname}_${version}-lunar_amd64.deb";
+      sha256 = "sha256-mpMJe8LAmO9OrqRIEWuxfVNeh5ANvjZIEHFz8cXYObY=";
+    };
+
+    buildInputs = [
+      nss
+      libvorbis
+      libdrm
+      libGL
+      wayland
+      libthai
+    ];
+
+    nativeBuildInputs = [ autoPatchelfHook dpkg ];
+
+    unpackPhase = ''
+      dpkg-deb --fsys-tarfile $src | tar -x --no-same-permissions --no-same-owner
+    '';
+
+    preBuild = ''
+      addAutoPatchelfSearchPath $out/lib/insync/
+    '';
+
+    installPhase = ''
+      runHook preInstall
+
+      mkdir -p $out/bin $out/lib $out/share
+      cp -R usr/* $out/
+
+      # use system glibc
+      rm -f $out/lib/insync/libgcc_s.so.1
+      rm -f $out/lib/insync/libstdc++.so.6
+
+      # remove badly packaged plugins
+      rm -f $out/lib/insync/PySide2/plugins/platforminputcontexts/libqtvirtualkeyboardplugin.so
+
+      runHook postInstall
+    '';
+
+    dontConfigure = true;
+    dontBuild = true;
+
+    # NB! This did the trick, otherwise it segfaults! However I don't understand why!
+    dontStrip = true;
+  };
+in buildFHSEnv { # ref: pkgs/build-support/build-fhsenv-bubblewrap/default.nix
+  name = "${pname}-${version}";
+  inherit meta;
+
+  # for including insync's xdg data dirs
+  extraOutputsToInstall = [ "share" ];
+
+  targetPkgs = pkgs: [
+    insync-pkg
+  ];
+
+  multiPkgs = pkgs: with pkgs; [
+    # apparently only package needed for the FHS :)
+    libudev0-shim
+  ];
+
+  runScript = writeShellScript "insync-wrapper.sh" ''
+    export QT_STYLE_OVERRIDE=Fusion
+    exec "${insync-pkg.outPath}/lib/insync/insync" "$@"
+  '';
+  # "insync start" command starts a daemon.
+  dieWithParent = false;
```

```suggestion
in buildFHSEnv {
  name = "${pname}-${version}";
  inherit pname version;
  
  # for including insync's xdg data dirs
  extraOutputsToInstall = [ "share" ];

  targetPkgs = pkgs: [
    insync-pkg
  ];

  multiPkgs = pkgs: with pkgs; [
    libudev0-shim
  ];

  runScript = writeShellScript "insync-wrapper.sh" ''
    export QT_STYLE_OVERRIDE=Fusion
    exec "${insync-pkg.outPath}/lib/insync/insync" "$@"
  '';

  # "insync start" command starts a daemon.
  dieWithParent = false;
  
  inherit meta;
```
Do we really must enforce the qt style?
Do we need to use multiPkgs aka 32 and 64 bit?

---

> Do we really must enforce the qt style?

Not necessarily, it was added to suppress one warning from Qt output.

>  Do we need to use multiPkgs aka 32 and 64 bit?

No, only 64bit binaries are shipped from the vendor.